cli/src/pcluster/config/cluster_config.py

```diff
@@ -1878,6 +1879,18 @@ def _register_validators(self):
             if self.scheduler_definition.requirements
             else None,
         )
+        system_users = get_attr(self.scheduler_definition, "system_users")
+        if system_users:
+            sudoer_configuration = [
+                get_attr(user, "sudoer_configuration")
+                for user in system_users
+                if get_attr(user, "sudoer_configuration")
+            ]
```

Comment:
Shall we move this entire logic in the validator and rather than passing sudoer_configuration to it we pass system_uses?

Reply:
Passing the logic into the validator will make it more complicate with unit test. In this case we need to create Systemusers config object in unit test

Reply:
Changed